vault/identity_store_oidc_provider_test.go

```diff
@@ -10,8 +10,576 @@ import (
+					Assignments:  []string{"test-assignment"},
+					Key:          "test-key",
+				},
+				authorizeRequest: &logical.Request{
+					Path:      "oidc/provider/test-provider/authorize",
+					Operation: logical.UpdateOperation,
+					Data: map[string]interface{}{
+						"client_id":     "",
+						"scope":         "openid",
+						"redirect_uri":  "https://localhost:8251/callback",
+						"response_type": "code",
+						"state":         "abcdefg",
+						"nonce":         "hijklmn",
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid authorize request using update operation (HTTP PUT/POST)",
+			args: args{
+				provider: new(provider),
+				assignment: &assignment{
+					Entities: []string{"test-entity"},
+				},
+				client: &client{
+					RedirectURIs: []string{"https://localhost:8251/callback"},
+					Assignments:  []string{"test-assignment"},
+					Key:          "test-key",
+				},
+				authorizeRequest: &logical.Request{
+					Path:      "oidc/provider/test-provider/authorize",
+					Operation: logical.UpdateOperation,
+					Data: map[string]interface{}{
+						"client_id":     "",
+						"scope":         "openid",
+						"redirect_uri":  "https://localhost:8251/callback",
+						"response_type": "code",
+						"state":         "abcdefg",
+						"nonce":         "hijklmn",
+					},
+				},
+			},
+		},
+		{
+			name: "valid authorize request using read operation (HTTP GET)",
+			args: args{
+				provider: new(provider),
+				assignment: &assignment{
+					Entities: []string{"test-entity"},
+				},
+				client: &client{
+					RedirectURIs: []string{"https://localhost:8251/callback"},
+					Assignments:  []string{"test-assignment"},
+					Key:          "test-key",
+				},
+				authorizeRequest: &logical.Request{
+					Path:      "oidc/provider/test-provider/authorize",
+					Operation: logical.ReadOperation,
+					Data: map[string]interface{}{
+						"client_id":     "",
+						"scope":         "openid",
+						"redirect_uri":  "https://localhost:8251/callback",
+						"response_type": "code",
+						"state":         "abcdefg",
+						"nonce":         "hijklmn",
+					},
+				},
+			},
+		},
+		{
+			name: "valid authorize request using client assignment with group membership",
+			args: args{
+				provider: new(provider),
+				assignment: &assignment{
+					Groups: []string{"test-group"},
+				},
+				client: &client{
+					RedirectURIs: []string{"https://localhost:8251/callback"},
+					Assignments:  []string{"test-assignment"},
+					Key:          "test-key",
+				},
+				authorizeRequest: &logical.Request{
+					Path:      "oidc/provider/test-provider/authorize",
+					Operation: logical.UpdateOperation,
+					Data: map[string]interface{}{
+						"client_id":     "",
+						"scope":         "openid",
+						"redirect_uri":  "https://localhost:8251/callback",
+						"response_type": "code",
+						"state":         "abcdefg",
+						"nonce":         "hijklmn",
+					},
+				},
+			},
+		},
```

Comment:
Adding test coverage for `max_age` parameter.

Reply:
I don't see test for `max_age`, is this a todo note for yourself?

Reply:
Yep, this is a todo for myself. Adding coverage for max_age.

Reply:
Added max_age coverage in 290fbaa.